### ISSUE TITLE
[CLOSED] QA-SRL

### ISSUE THREAD

**Issue by jeswan** [nyu-mll/jiant-v1-legacy#716]
<a href="https://github.com/zphang"><img src="https://avatars1.githubusercontent.com/u/1668462?v=4" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [zphang](https://github.com/zphang)**
_Friday Jun 21, 2019 at 12:19 GMT_
_Originally opened as https://github.com/nyu-mll/jiant/pull/716_

----

Re: #571

Starting with a basic implementation of QA-SRL, seeking feedback.

- There is quite a lot of repeated/branching input sets in QA-SRL. Each sentence has multiple "verbs", each verb corresponds to multiple questions, each question has multiple answers from different annotators, and each answer contains potentially more than one span. Currently I'm doing a naive implementation of enumerating *all* of these. For a sense of scale, there are about 44k sentences in the training set, but about 600k+ total training examples when expanded.
- The QA-SRL / Large-Scale QA-SRL papers did not talk too much about scoring, and where they did I'm not sure if it works best for pretraining. I'm currently implementing the following:
- **Training loss**: I am currently following the BERT formulation for SQuAD of straightforwardly predicting the span start and ends via a softmax over the number of tokens in the input, with a cross-entropy loss. No correction for ensuring start < end.
- **Evaluation**: I am currently using the SQuAD-style macro-F1, which treats the predicted / gold tokens as individual observations, computing the F1 for each example, and then averaging across examples.

----

_**[zphang](https://github.com/zphang)** included the following code: https://github.com/nyu-mll/jiant/pull/716/commits_
